src/System.Private.DataContractSerialization/src/System/Xml/XmlBaseReader.cs

```diff
@@ -832,6 +832,12 @@ public override XmlNodeType MoveToContent()
                 {
                     if (_node.NodeType != XmlNodeType.Text && _node.NodeType != XmlNodeType.CDATA)
                         break;
+
+                    if (_trailByteCount > 0)
+                    {
+                        break;
+                    }
+
```

Comment:
Maybe @khdang can give any insights because this is going to be the second change in this file since the initial commit by @khdang. See history: https://github.com/dotnet/corefx/commits/master/src/System.Private.DataContractSerialization/src/System/Xml/XmlBaseReader.cs.